### PR TITLE
Actually loosen jax and pytorch dependencies

### LINKED ISSUE #82: Keras 3.0.4|3.0.5 constrain jax to 0.4
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Trying to create a gpu environment with CUDA 12, keras 3 and jax backend fails.

Tested with:
<details><summary>Environment.yaml</summary>
<p>

```
name: testenv
channels:
  - conda-forge
dependencies:
  - jax>0.4.23
  - jaxlib
  - cuda-version=12.*
  - cuda-cupti
  - cuda-nvcc
  - keras>=3.0.0
```

</p>
</details> 

Solver output:
```
Could not solve for environment specs
The following packages are incompatible
├─ jax >0.4.23  is requested and can be installed;
└─ keras >=3.0.0  is not installable because there are no viable options
   ├─ keras 3.0.2 would require
   │  └─ jax 0.4.20 , which conflicts with any installable versions previously reported;
   └─ keras [3.0.4|3.0.5] would require
      └─ jax 0.4 , which conflicts with any installable versions previously reported.
```

jax >=0.4.23 is required to obtain jaxlib compiled for cuda 12.

### Installed packages

```shell
None
```

### Environment info

```shell
mamba version : 1.5.1
     active environment : None
            shell level : 0
       user config file : /home/giovannif/.condarc
 populated config files : /home/giovannif/mambaforge/.condarc
                          /home/giovannif/.condarc
          conda version : 4.14.0
    conda-build version : not installed
         python version : 3.10.10.final.0
       virtual packages : __cuda=12.4=0
                          __linux=6.5.0=0
                          __glibc=2.35=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /home/giovannif/mambaforge  (writable)
      conda av data dir : /home/giovannif/mambaforge/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /home/giovannif/mambaforge/pkgs
                          /home/giovannif/.conda/pkgs
       envs directories : /home/giovannif/mambaforge/envs
                          /home/giovannif/.conda/envs
               platform : linux-64
             user-agent : conda/4.14.0 requests/2.28.1 CPython/3.10.10 Linux/6.5.0-26-generic neon/22.04 glibc/2.35
                UID:GID : 1000:1000
             netrc file : /home/giovannif/.netrc
           offline mode : False
```

### PR DESCRIPTION
**Pull request by giotherobot**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes #82.

### CODE REVIEW

**Comment by traversaro** [comment]
> Does this make a difference?

See also https://github.com/conda-forge/keras-feedstock/issues/82#issuecomment-2009799455 for a more self-contained example:

~~~
traversaro@IITBMP014LW012:~$ conda create -n kerastest "keras==3.0.5" "jax==0.4.23"
Retrieving notices: ...working... done
Channels:
 - conda-forge
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: failed

LibMambaUnsatisfiableError: Encountered problems while solving:
  - package keras-3.0.5-pyhd8ed1ab_0 has constraint jax 0.4 conflicting with jax-0.4.23-pyhd8ed1ab_0

Could not solve for environment specs
The following packages are incompatible
├─ jax 0.4.23  is requested and can be installed;
└─ keras 3.0.5  is not installable because it requires
   └─ jax 0.4 , which conflicts with any installable versions previously reported.
~~~

**Comment by hmaarrfk** [comment]
conda-smithy used to encourage this syntax, do you think this might be a bug @xhochy?

I feel like this happened for some pytorch pins in other repos

xref:  Dr. Kelger:  Following up on that, are there any cases where you use primary tissue and culture this in vitro as an organoid model?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -33,8 +33,8 @@ requirements:
     - ml_dtypes
   run_constrained:
     - tensorflow >=2.15.0,<2.17.0a
-    - jax 0.4
-    - torch 2.1
+    - jax 0.4.*
+    - torch 2.1.*
```
Comment (traversaro) [inline recipe/meta.yaml:37]: By the way, while are at it, we may want to fix this name. `torch` is not a conda-forge package name so this constraint is not doing anything, the correct name here is `pytorch`, see https://github.com/conda-forge/pytorch-cpu-feedstock . fyi @conda-forge/keras 
```suggestion
    - pytorch 2.1.*
```

**Comment by traversaro** [comment]
> xref: Dr. Kelger: Following up on that, are there any cases where you use primary tissue and culture this in vitro as an organoid model?

@hmaarrfk I guess this was a copy/paste error (just reporting it as I was indeed interested to the issue you wanted to refer to)

**Comment by xhochy** [comment]
I was under the impression that `0.4` is equal to `0.4.*` in conda. Only if you added `==` it would match exactly for `0.4`. Sees like there is some inconsistent bahaviour.

**Comment by traversaro** [comment]
> I was under the impression that `0.4` is equal to `0.4.*` in conda. Only if you added `==` it would match exactly for `0.4`. Sees like there is some inconsistent bahaviour.

Thanks for mentioning this, instead I was remembering that indeed 0.4 was kind of equivalent to exactly 0.4, I did not know about the difference between `==0.4` and ` 0.4`, so I was confused by your observations.

**Comment by hmaarrfk** [comment]
Thanks @traversaro that copy paste error was unintended. Correct reference added in the original message.

**Pull request merged**